src/aws-adapter.js

```diff
@@ -135,19 +139,24 @@ async function lambdaAdapter(event, context, secrets = {}) {
  */
 async function lambda(evt, ctx) {
   try {
+    evt.nonHttp = (!evt.requestContext);
+
     const secrets = await getAWSSecrets(ctx.functionName);
     let handler = (event, context) => lambdaAdapter(event, context, secrets);
-    if (secrets.EPSAGON_TOKEN) {
-      // check if health check
-      const suffix = evt.pathParameters && evt.pathParameters.path ? `/${evt.pathParameters.path}` : '';
-      if (suffix !== HEALTHCHECK_PATH) {
-        handler = epsagon(handler, {
-          token: secrets.EPSAGON_TOKEN,
-        });
+
+    if (!evt.nonHttp) {
+      // do not use Epsagon otherwise as it currently reports issues for functions that throw
+      if (secrets.EPSAGON_TOKEN) {
+        // check if health check
+        const suffix = evt.pathParameters && evt.pathParameters.path ? `/${evt.pathParameters.path}` : '';
+        if (suffix !== HEALTHCHECK_PATH) {
+          handler = epsagon(handler, {
+            token: secrets.EPSAGON_TOKEN,
+          });
+        }
```

Comment:
I think we should keep the espagon for all ases

Reply:
What do you suggest: whenever we deploy a new function triggered by SQS that returns a temporary failure for a message, we need to go to Epsagon to mute the issue (and ignore the alerts in the escalation channel)?

Reply:
yes.